src/FlowDiagnostics.jl

```diff
@@ -315,4 +316,104 @@ function DirectionalErtelPotentialVorticity(model, direction; location = (Face,
 end
 #---
 
+#+++ Velocity gradient tensor
+@inline fψ_plus_gφ²(i, j, k, grid, f, ψ, g, φ) = @inbounds (f(i, j, k, grid, ψ) + g(i, j, k, grid, φ))^2
+
+function strain_rate_tensor_modulus_ccc(i, j, k, grid, u, v, w)
+    Sˣˣ² = ∂xᶜᶜᶜ(i, j, k, grid, u)^2
+    Sʸʸ² = ∂yᶜᶜᶜ(i, j, k, grid, v)^2
+    Sᶻᶻ² = ∂zᶜᶜᶜ(i, j, k, grid, w)^2
+
+    Sˣʸ² = ℑxyᶜᶜᵃ(i, j, k, grid, fψ_plus_gφ², ∂yᶠᶠᶜ, u, ∂xᶠᶠᶜ, v) / 4
+    Sˣᶻ² = ℑxzᶜᵃᶜ(i, j, k, grid, fψ_plus_gφ², ∂zᶠᶜᶠ, u, ∂xᶠᶜᶠ, w) / 4
+    Sʸᶻ² = ℑyzᵃᶜᶜ(i, j, k, grid, fψ_plus_gφ², ∂zᶜᶠᶠ, v, ∂yᶜᶠᶠ, w) / 4
+
+    return √(Sˣˣ² + Sʸʸ² + Sᶻᶻ² + 2 * (Sˣʸ² + Sˣᶻ² + Sʸᶻ²))
+end
+
+"""
+    $(SIGNATURES)
+
+Calculate the modulus (absolute value) of the strain rate tensor `S`, which is defined as the
+symmetric part of the velocity gradient tensor:
+
+    Sᵢⱼ = ½(∂ⱼuᵢ + ∂ᵢuⱼ)
+
+Its modulus is then defined (using Einstein summation notation) as
+
+    || Sᵢⱼ || = √( Sᵢⱼ Sᵢⱼ)
+
+"""
+function StrainRateTensorModulus(model; location = (Center, Center, Center))
+    validate_location(location, "StrainRateTensorModulus", (Center, Center, Center))
+    return KernelFunctionOperation{Center, Center, Center}(strain_rate_tensor_modulus_ccc, model.grid, model.velocities...)
+end
+
+
+@inline fψ_minus_gφ²(i, j, k, grid, f, ψ, g, φ) = @inbounds (f(i, j, k, grid, ψ) - g(i, j, k, grid, φ))^2
```

Comment:
```suggestion
@inline fψ_minus_gφ²(i, j, k, grid, f, ψ, g, φ) = (f(i, j, k, grid, ψ) - g(i, j, k, grid, φ))^2
```

Reply:
Just so that I understand, the reasoning here is that since there isn't a call to `getindex` here `@inbounds` is unnecessary?

Reply:
Yes, `i, j, k` are just args in a function. There is no `getindex` involved.